flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/JoinToMultiJoinRule.java

```diff
@@ -723,6 +737,71 @@ private boolean containsSnapshot(RelNode relNode) {
         }
     }
 
+    /**
+     * Processes state TTL hints for a given side (left or right) of a join operation.
```

Comment:
nit: instead of processes, could we detail what this method does - maybe rename the method to indicate what it does . Maybe something like addSideHintstoHintBuilder